aiida/backends/sqlalchemy/migrations/versions/91b573400be5_prepare_schema_reset.py

```diff
@@ -40,8 +40,8 @@ def upgrade():
     # isolated way, we do not suffer from those problems and can safely use it.
     statement = text(
         """
-        INSERT INTO db_dbsetting (key, val, description)
-        SELECT 'schema_generation', '"1"', 'Database schema generation'
+        INSERT INTO db_dbsetting (key, val, description, time)
+        SELECT 'schema_generation', '"1"', 'Database schema generation', NOW()
```

Comment:
Well it already is done, in `aiida/backends/sqlalchemy/migrations/versions/1de112340b16_django_parity_1.py`, that sets `null` to the current time.
Off-hand I can't remember why I also added it here as well, I guess it was breaking a test but will check

Reply:
I am not sure if this is the cause, but when I run the migrations of your branch on an existing SQLA profile, the first one gets applied, but then it seems to get stuck at the second one. I have to kill the postgres process and then get the following stacktrace:
```
Traceback (most recent call last):
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
psycopg2.OperationalError: SSL SYSCALL error: EOF detected


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/sph/.virtualenvs/aiida_dev/bin/verdi", line 12, in <module>
    sys.exit(verdi())
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/cmdline/commands/cmd_storage.py", line 42, in storage_migrate
    backend.migrate()
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/orm/implementation/sqlalchemy/backend.py", line 44, in migrate
    self._schema_manager.migrate()
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/manager.py", line 157, in migrate
    self._migrate_database_version()
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/sqlalchemy/manager.py", line 160, in _migrate_database_version
    self.migrate_up('head')
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/sqlalchemy/manager.py", line 147, in migrate_up
    upgrade(config, version)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/command.py", line 320, in upgrade
    script.run_env()
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/script/base.py", line 563, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 92, in load_python_file
    module = load_module_py(module_id, path)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 108, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/sqlalchemy/migrations/env.py", line 52, in <module>
    run_migrations_online()
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/sqlalchemy/migrations/env.py", line 45, in run_migrations_online
    context.run_migrations()  # pylint: disable=no-member
  File "<string>", line 8, in run_migrations
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/runtime/environment.py", line 851, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/runtime/migration.py", line 612, in run_migrations
    step.migration_fn(**kw)
  File "/home/sph/code/aiida/env/dev/aiida-core/aiida/backends/sqlalchemy/migrations/versions/1de112340b17_django_parity_2.py", line 80, in upgrade
    op.alter_column('db_dbsetting', 'time', existing_type=sa.DateTime(timezone=True), nullable=False)
  File "<string>", line 8, in alter_column
  File "<string>", line 3, in alter_column
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/operations/ops.py", line 1880, in alter_column
    return operations.invoke(alt)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/operations/base.py", line 387, in invoke
    return fn(self, operation)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/operations/toimpl.py", line 50, in alter_column
    operations.impl.alter_column(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/ddl/postgresql.py", line 173, in alter_column
    super(PostgresqlImpl, self).alter_column(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/ddl/impl.py", line 231, in alter_column
    self._exec(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/alembic/ddl/impl.py", line 197, in _exec
    return conn.execute(construct, multiparams)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/future/engine.py", line 280, in execute
    return self._execute_20(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1614, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/sql/ddl.py", line 80, in _execute_on_connection
    return connection._execute_ddl(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1381, in _execute_ddl
    ret = self._execute_context(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1845, in _execute_context
    self._handle_dbapi_exception(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2026, in _handle_dbapi_exception
    util.raise_(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/home/sph/.virtualenvs/aiida_dev/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) SSL SYSCALL error: EOF detected

[SQL: ALTER TABLE db_dbsetting ALTER COLUMN time SET NOT NULL]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```
The final error message seems interesting. It may be coincidence, but it is exactly concerning the time column of `db_dbsetting`.

Reply:
Just tried this on an empty database and it also hangs completely. Surely it cannot take more than 5 minutes on an empty database, so it must be a problem.

Reply:
What if you undo this change, or comment everything out in the aiida/backends/sqlalchemy/migrations/versions/1de112340b17_django_parity_2.py?
As discussed I had issues with "locked" databases both here, before fixing the issue that separate migrations were not being run in separate transactions, and in #5192, before fixing a separate sqlalchemy session not being closed.

Reply:
this has been removed